packages/anu-vue/src/components/loader/ALoader.vue

```diff
@@ -0,0 +1,118 @@
+<script setup lang="ts">
+import { ATypography } from '../typography'
+import ASpinner from './ASpinner.vue'
+import { loaderProps } from './props'
+import { isTypographyUsed } from '@/components/typography/utils'
+import { ConfigurableValue, useConfigurable } from '@/composables/useConfigurable'
+import { useLayer } from '@/composables/useLayer'
+
+const props = defineProps(loaderProps)
+
+defineOptions({
+  name: 'ALoader',
+})
+
+const slots = useSlots()
+const attrs = useAttrs()
+
+const loading = toRef(props, 'loading')
+
+const { getLayerClasses } = useLayer()
+const { styles, classes } = getLayerClasses(
+  toRef(props, 'color'),
+  toRef(props, 'variant'),
+  toRef(props, 'states'),
+)
+
+// FIXME Can't found a better way to change the background overlay depending of the layer. It's not perfect for transparent variant like light, text or outline.
+const attrsClasses = attrs.class as string || ''
+const overlayClasses = computed(() => {
+  // We skip if there is some bg-* classes
+  if (!props.overlay || attrsClasses.split(' ').some(c => /^bg-(.+)$/.test(c)))
+    return
+  if (!props.variant || props.variant === 'fill')
+    return 'bg-$a-layer-color'
+
+  if (props.variant === 'outline')
+    return 'bg-white dark:bg-dark !border-none states'
+
+  if (props.variant === 'light')
+    return 'bg-$a-layer-color states'
+
+  return 'bg-$a-layer-color' // bg-light dark:bg-dark'
+})
+
+if (props.overlay) {
+  onMounted(() => {
+    const instance = getCurrentInstance()
+
+    // Add `relative` class to parent if overlay props
+    // FIXME It's work but it's ugly and not secure at all. Should we ask developer to add `relative` class themselves ?
+    if (instance?.parent?.vnode.el)
+      instance.parent.vnode.el.classList.add('relative', 'overflow-hidden')
+  })
+}
+
+// Prevent scrolling when full page mode
+if (props.fullPage) {
+  const isLocked = useScrollLock(document.body, false)
+
+  watch(loading, () => {
+    isLocked.value = !!loading.value
+  }, { immediate: true })
+}
+
+const _isTypographyUsed = isTypographyUsed(toRefs(props), slots)
+
+// Modify text prop to have `text-sm`
+const _textProp = useConfigurable(toRef(props, 'text'))
+if (_textProp.value.classes === undefined)
+  _textProp.value.classes = 'text-sm'
+else if (Array.isArray(_textProp.value.classes))
+  _textProp.value.classes = [..._textProp.value.classes, 'text-sm']
+else
+  _textProp.value.classes = ' text-sm'
+</script>
+
+<template>
+  <span
+    v-if="loading"
+    class="a-loader-overlay"
+    :class="[
+      (overlay || fullPage) && 'absolute z-1 top-0 bottom-0 left-0 right-0 w-full flex flex-col gap-2 items-center justify-center text-center',
+      fullPage && 'fixed !z-9999',
+      overlayClasses,
+      ...classes,
+    ]"
+    :style="styles"
+  >
+    <!-- 👉 Slot: Default -->
+    <slot>
+      <ASpinner class="a-loader-spinner inline w-$a-spinner-size h-$a-spinner-size animate-spin" />
```

Comment:
I added customization :

![image](https://user-images.githubusercontent.com/188172/211714710-2614622a-b0d4-4598-b5a9-d594101ec1ab.png)

I even don't know what to do with the ASpinner component... As I said, it could be awesome to configure the spinner global. But how to do that ?

In fact, I've started another branch with utilities to work with svg, circle and path. I experimented a lot and it's really not finish. But I created knob, chart and of course spinners !
OK, I'm going to push it, even if it's not finished.